urbit/app/eth-event-viewer.hoon

```diff
@@ -0,0 +1,409 @@
+/-  ew-sur=eth-watcher, *eth-event-viewer
+/+  *server, default-agent, dbug, ew-lib=eth-watcher, eev=eth-event-viewer
+/=  index
+  /^  octs
+  /;  as-octs:mimes:html
+  /:  /===/app/eth-event-viewer/index
+  /|  /html/
+      /~  ~
+  ==
+/=  tile-js
+  /^  octs
+  /;  as-octs:mimes:html
+  /:  /===/app/eth-event-viewer/js/tile
+  /|  /js/
+      /~  ~
+  ==
+/=  script
+  /^  octs
+  /;  as-octs:mimes:html
+  /:  /===/app/eth-event-viewer/js/index
+  /|  /js/
+      /~  ~
+  ==
+/=  style
+  /^  octs
+  /;  as-octs:mimes:html
+  /:  /===/app/eth-event-viewer/css/index
+  /|  /css/
+      /~  ~
+  ==
+/=  eth-event-viewer-png
+  /^  (map knot @)
+  /:  /===/app/eth-event-viewer/img  /_  /png/
+=,  format
+::
+|%
++$  card  card:agent:gall
+::
++$  versioned-state
+  $%  state-zero
+  ==
++$  state-zero  [%0 contracts=(map @ux contract-type)]
+--
+=|  state-zero
+=*  state  -
+^-  agent:gall
+=<
+  |_  bol=bowl:gall
+  +*  this       .
+      eth-event-viewer-core  +>
+      cc         ~(. eth-event-viewer-core bol)
+      def        ~(. (default-agent this %|) bol)
+  ::
+  ++  on-init
+    ^-  (quip card _this)
+    =/  eev  /eth-event-viewer
+    =/  tile-js  '/~eth-event-viewer/js/tile.js'
+    =/  launcha  [%launch-action !>([%add %eth-event-viewer / tile-js])]
```

Comment:
These paths were mainly created by the landscape template but I can edit them.

What do you mean by:
"Would be nice if this also read from the data.event-log field and rendered those in the event lists too. It already loads in full event signatures, so should be able to do that, right?"

Reply:
In the UI, iirc you currently only see the _topics_ of the event, rather than all of the event's values. The non-topic values are encoded (with, uh, RLP encoding) in `data.event-log`.

Reply:
ah now I understand. do you want to display it decoded? if yes, how I am doing that ? I've tried to use https://github.com/ethereumjs/rlp but wasn't able to decode the hex

Reply:
Since (I believe) you know the event's function signature, maybe you can just extract the relevant `etyp:abi`s from that and then use `+decode-results:abi` to decode into `data:abi`, which wouldn't be hard to turn into json and send to the frontend.

Reply:
I've made another try using this package but I am not able to serve the app properly:
https://www.npmjs.com/package/ethereum-input-data-decoder

`'fs' is imported by node_modules/xmlhttprequest/lib/XMLHttpRequest.js, but could not be resolved – treating it as an external dependency
'crypto' is imported by node_modules/uuid/rng.js, but could not be resolved – treating it as an external dependency
'child_process' is imported by node_modules/xmlhttprequest/lib/XMLHttpRequest.js, but could not be resolved – treating it as an external dependency
'child_process' is imported by commonjs-external-child_process, but could not be resolved – treating it as an external dependency
'url' is imported by node_modules/xmlhttprequest/lib/XMLHttpRequest.js, but could not be resolved – treating it as an external dependency
'url' is imported by commonjs-external-url, but could not be resolved – treating it as an external dependency
'http' is imported by node_modules/xmlhttprequest/lib/XMLHttpRequest.js, but could not be resolved – treating it as an external dependency
'http' is imported by commonjs-external-http, but could not be resolved – treating it as an external dependency
'https' is imported by node_modules/xmlhttprequest/lib/XMLHttpRequest.js, but could not be resolved – treating it as an external dependency
'https' is imported by commonjs-external-https, but could not be resolved – treating it as an external dependency
'crypto' is imported by node_modules/brorand/index.js, but could not be resolved – treating it as an external dependency
'fs' is imported by node_modules/bindings/bindings.js, but could not be resolved – treating it as an external dependency
'buffer' is imported by node_modules/safe-buffer/index.js, but could not be resolved – treating it as an external dependency
'path' is imported by node_modules/bindings/bindings.js, but could not be resolved – treating it as an external dependency
'path' is imported by commonjs-external-path, but could not be resolved – treating it as an external dependency
'util' is imported by node_modules/inherits/inherits.js, but could not be resolved – treating it as an external dependency
'util' is imported by commonjs-external-util, but could not be resolved – treating it as an external dependency
'path' is imported by node_modules/file-uri-to-path/index.js, but could not be resolved – treating it as an external dependency
'stream' is imported by node_modules/keccak/lib/api/keccak.js, but could not be resolved – treating it as an external dependency
'stream' is imported by node_modules/keccak/lib/api/shake.js, but could not be resolved – treating it as an external dependency
'stream' is imported by commonjs-external-stream, but could not be resolved – treating it as an external dependency
'buffer' is imported by node_modules/keccak/node_modules/safe-buffer/index.js, but could not be resolved – treating it as an external dependency
[19:02:46] Starting 'urbit-copy'...
[19:02:46] Finished 'urbit-copy' after 45 ms
`

do you have an idea why I am having this issue?

Reply:
used import {InputDataDecoder} from 'ethereum-input-data-decoder';

Reply:
ok as discussed this will be implemented in a future bounty since it would be too much for this one to implement